packages/grafana-runtime/src/services/live.ts

```diff
@@ -20,68 +16,47 @@ export interface LiveDataFilter {
 }
 
 /**
- * @alpha
+ * Indicate if the frame is appened or replace
+ *
+ * @public -- but runtime
  */
-export interface LiveDataStreamOptions {
-  addr: LiveChannelAddress;
-  frame?: DataFrame; // initial results
-  key?: string;
-  buffer?: StreamingFrameOptions;
-  filter?: LiveDataFilter;
+export enum StreamingFrameAction {
+  Append = 'append',
+  Replace = 'replace',
 }
 
 /**
- * @alpha -- experimental
+ * @alpha
  */
-export enum StreamingResponseDataType {
-  NewValuesSameSchema = 'NewValuesSameSchema',
-  FullFrame = 'FullFrame',
+export interface StreamingFrameOptions {
+  maxLength: number; // 1000
+  maxDelta: number; // how long to keep things
+  action: StreamingFrameAction; // default will append
 }
 
 /**
- * @alpha -- experimental
+ * @alpha
  */
-export type StreamingResponseDataTypeToData = {
-  [StreamingResponseDataType.NewValuesSameSchema]: {
-    values: unknown[][];
+export function getStreamingFrameOptions(opts?: Partial<StreamingFrameOptions>): StreamingFrameOptions {
+  return {
+    maxLength: opts?.maxLength ?? 1000,
+    maxDelta: opts?.maxDelta ?? Infinity,
+    action: opts?.action ?? StreamingFrameAction.Append,
   };
-  [StreamingResponseDataType.FullFrame]: {
-    frame: ReturnType<StreamingDataFrame['serialize']>;
-  };
-};
-
-/**
- * @alpha -- experimental
- */
-export type StreamingResponseData<T = StreamingResponseDataType> = T extends StreamingResponseDataType
-  ? {
-      type: T;
-    } & StreamingResponseDataTypeToData[T]
-  : never;
-
-/**
- * @alpha -- experimental
- */
-export const isStreamingResponseData = <T extends StreamingResponseDataType>(
-  responseData: DataQueryResponseData,
-  type: T
-): responseData is StreamingResponseData<T> => 'type' in responseData && responseData.type === type;
-
-const AllStreamingResponseDataTypes = Object.values(StreamingResponseDataType);
+}
 
 /**
- * @alpha -- experimental
+ * @alpha
  */
-export const isAnyStreamingResponseData = (
-  responseData: DataQueryResponseData
-): responseData is StreamingResponseData =>
-  'type' in responseData && AllStreamingResponseDataTypes.includes(responseData.type);
+export interface LiveDataStreamOptions {
+  addr: LiveChannelAddress;
+  frame?: DataFrame; // initial results
+  key?: string;
+  buffer?: StreamingFrameOptions;
+  filter?: LiveDataFilter;
+}
 
-/**
- * @alpha -- experimental
- */
-export const isStreamingDataFrame = (data: DataQueryResponseData): data is StreamingDataFrame =>
-  isDataFrame(data) && 'packetInfo' in data;
+type StreamingResponseData = any; // ????
```

Comment:
@ArturWierzbicki -- this is the interface/location I don't know how to deal with :cry: 

If we can keep specifics out of this, i think we will be happier